hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java

```diff
@@ -230,12 +232,25 @@ private void mapModelChanges(Token token, MerkleToken mutableToken) {
         mutableToken.setExpiry(token.getExpiry());
     }
 
-    private void mapModelChanges(UniqueToken nft, MerkleUniqueToken mutableNft) {
-        mutableNft.setOwner(nft.getOwner().asEntityId());
-        mutableNft.setSpender(nft.getSpender().asEntityId());
-        mutableNft.setMetadata(nft.getMetadata());
-        final var creationTime = nft.getCreationTime();
-        mutableNft.setPackedCreationTime(
-                packedTime(creationTime.getSeconds(), creationTime.getNanos()));
+    private void mapModelChanges(UniqueToken nft, UniqueTokenAdapter mutableNft) {
+        if (mutableNft.isVirtual()) {
```

Comment:
Is there a reason to not just use the `UniqueTokenAdapter` top-level methods here?